packages/ember-simple-auth/addon/mixins/application-route-mixin.js

```diff
@@ -5,9 +5,16 @@ import { A } from '@ember/array';
 import { getOwner } from '@ember/application';
 import { inject } from '@ember/service';
 import Ember from 'ember';
+import { deprecate } from '@ember/application/deprecations';
 import Configuration from './../configuration';
+
 import isFastBoot from 'ember-simple-auth/utils/is-fastboot';
 
+deprecate('Ember Simple Auth: The ApplicationRouteMixin is now deprecated; it can be safely removed.', false, {
+  id: 'ember-simple-auth.mixins.application-route-mixin',
+  until: '4.0.0'
+});
```

Comment:
We should add a reference to the README step to help users know where to look for instructions on the changes they need to make. As it stands the message can be understood as if one could simply remove the `ApplicationRouteMixin` without the need for further steps.

Reply:
That is actually the case.

Reply:
Would they not need to use the session service within their `beforeModel` hooks in order to achieve parity with the mixin usage?

Reply:
No, the initializer does everything automatically now – see https://github.com/simplabs/ember-simple-auth/pull/2198/files#diff-db561403ef670c68769b736f6ccfacfcR14